yb-voyager/cmd/importData.go

```diff
@@ -855,39 +858,49 @@ func doOneImport(task *SplitFileImportTask, connPool *tgtdb.ConnectionPool) {
 					}
 				}
 				res, err := conn.PgConn().CopyFrom(context.Background(), reader, copyCommand)
-				rowsCount = res.RowsAffected()
+				rowsAffected = res.RowsAffected()
 
+				if err != nil && (strings.Contains(err.Error(), "invalid input syntax") ||
+					strings.Contains(err.Error(), "violates unique constraint")) {
+					return false, err
+				}
+
+				/*
+					Note: If a user retries after deleting some row(s) from a batch,
+					yb-voyager will never be able to mark the batch as completed
+					github issue: https://github.com/yugabyte/yb-voyager/issues/223
+				*/
 				if err != nil {
 					log.Warnf("COPY FROM file %q: %s", inProgressFilePath, err)
-					if !strings.Contains(err.Error(), "violates unique constraint") {
+					if err != nil {
```

Comment:
This condition is redundant because of line 873.

Reply:
done